src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ConstantCallSite.cs

```diff
@@ -8,9 +8,9 @@ namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
     internal sealed class ConstantCallSite : ServiceCallSite
     {
         private readonly Type _serviceType;
-        internal object DefaultValue => Value;
+        internal object? DefaultValue => Value;
 
-        public ConstantCallSite(Type serviceType, object defaultValue): base(ResultCache.None)
+        public ConstantCallSite(Type? serviceType, object? defaultValue): base(ResultCache.None)
```

Comment:
When can `serviceType` ever be `null`? Also, this throws if you do pass in `null`, so it should be:

```suggestion
        public ConstantCallSite(Type serviceType, object? defaultValue): base(ResultCache.None)
```

Reply:
Good catch, thanks! Missed the `throw` somehow. Reverted the change.